designs/2021-stricter-rule-test-validation/README.md

```diff
@@ -0,0 +1,204 @@
+    burden, incompatibility with other tools, breaking existing user
+    experience, etc. Try to identify as many potential problems with
+    implementing this RFC as possible.
+-->
+
+Plugin authors who left out key properties from their test cases will see a slight increase in testing burden.
+
+## Backwards Compatibility Analysis
+
+<!--
+    How does this change affect existing ESLint users? Will any behavior
+    change for them? If so, how are you going to minimize the disruption
+    to existing users?
+-->
+
+This RFC describes a set of breaking changes that can be released in the next major version of ESLint. These changes will only affect ESLint plugin authors, and only when they perform the relevant major ESLint version upgrade internally in their repositories. Some plugin authors will need to add test coverage by filling in missing properties. Users of ESLint will not be affected.
+
+## Alternatives
+
+<!--
+    What other designs did you consider? Why did you decide against those?
+
+    This section should also include prior art, such as whether similar
+    projects have already implemented a similar feature.
+-->
+
+The alternative is not adding additional validation, allowing ESLint tests to be more of a free-for-all with varying levels of quality and coverage.
+
+I also considered whether we could make validation even stricter than what's proposed here. In terms of required vs. optional test case properties, I didn't see any other properties that I thought should become mandatory. `type`, `line`, `column`, `endLine`, and `endColumn` are optional error object properties that can be useful but aren't critical to include in all test cases.
+
+## Open Questions
+
+<!--
+    This section is optional, but is suggested for a first draft.
+
+    What parts of this proposal are you unclear about? What do you
+    need to know before you can finalize this RFC?
+
+    List the questions that you'd like reviewers to focus on. When
+    you've received the answers and updated the design to reflect them,
+    you can remove this section.
+-->
+
+There are two more potential ideas that I have left out of the RFC body because they could be considered more stylistic in nature. I am in favor of them, so if people are interested, I could elevate them to the RFC body (or they could be considered in a separate RFC).
+
+1. Disallow repeating the test case `code` as `output` for non-fixable test cases. Instead, either use `output: null` or omit `output` to clearly and concisely indicate that a test case produces no autofix. This is based on the [eslint-plugin/prefer-output-null](https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/blob/master/docs/rules/prefer-output-null.md) lint rule.
+2. Disallow identical test cases. This is based on the [eslint-plugin/no-identical-tests](https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/blob/master/docs/rules/no-identical-tests.md) lint rule.
```

Comment:
I’m also in favor of this.

Reply:
Sounds good, I'll move this to the RFC body. Note checking for identical test cases is more complex and so I would expect the implementation of it to be separate from the implementation of the other assertions being added.

Reply:
Moved to RFC body and expanded detail.